Fix radclient i option

bmork opened the pull request:
This is an attempt to fix issue #2194 

Lightly tested with radclient, but not yet with the server.

arr2036 commented on src/protocols/radius/list.c:
```diff
@@ -590,6 +593,18 @@ bool fr_packet_list_id_alloc(fr_packet_list_t *pl, int proto,
+		 */
+
+		if (id != -1) {
+			if  ((ps->id[(id >> 3) & 0x1f] & (1 << (id & 0x07))) != 0) continue;
```
Hmm what's the idea here?

alandekok replied:
The IDs are stored as a bitmask, so this is checking if the ID is allocated.

alandekok commented on src/bin/radclient.c:
```diff
@@ -855,7 +855,7 @@ static int send_one_packet(rc_request_t *request)
 	 *	Haven't sent the packet yet.  Initialize it.
 	 */
-	if (request->packet->id == -1) {
+	if (fr_packet_list_find(packet_list, request->packet) == NULL) {
```
We should really avoid expensive lookups when a simple flag would do.  It's better to add a flag, or use a flag in `request->packet`.

bmork replied:
OK.  I thought it didn't matter that much in radclient since the packet_list is short anyway. I'll try to make this cheaper in the next version.

bmork replied:
I started out adding a new "sent" boolean to the request, but noticed that "tries" already has this info.  Or is there something I'm missing now?  I've updated the branch with a new version

alandekok commented on src/protocols/radius/list.c:
```diff
@@ -508,7 +508,10 @@ bool fr_packet_list_id_alloc(fr_packet_list_t *pl, int proto,
 	 *	and near-zero cost on free.
 	 */
 
-	id = fd = -1;
```
There's no need to move this..

bmork replied:
Will fix

bmork commented:
ping?  I believe I addressed all the comments on this request.  Please close it, with or wthout merging as you see best - that's your privilege to decide :-)

Unless there was something I missed?

alandekok commented:
I think it should be OK.  I'll have to take a deeper look

If this was v3, I'd be worried about changing the server functionality, just to fix radclient.  But the proxying in "master" uses an entirely different API to allocate IDs.

bmork commented:
> I think it should be OK. I'll have to take a deeper look
> 
> If this was v3, I'd be worried about changing the server functionality, just to fix radclient. But the proxying in "master" uses an entirely different API to allocate IDs.

Fair enough.  I believe it should have no measurale impact on the server, but verifying this would be good.

arr2036 commented:
@bmork can you rebase and I'll merge this.

bmork commented:
Hope I got the rebasing right.  Didn't really remember writing this code ;-)
